robottelo/hosts.py

```diff
@@ -554,6 +660,8 @@ def remove_katello_ca(self):
         result = self.execute('rpm -qa |grep katello-ca-consumer')
         if result.status == 0:
             raise ContentHostError(f'katello-ca rpm(s) are still installed: {result.stdout}')
+        # unregister host from CDN to avoid subscription leakage
+        self.execute('subscription-manager unregister')
```

Comment:
Good move! 
We need to stop leaking subs but unregister needs to be performed before removing `katello-ca-consumer-*` rpm (line 657)